src/azul/__init__.py

```diff
@@ -582,6 +583,15 @@ def is_integration_test_catalog(self) -> bool:
                 require(self.internal, 'IT catalogs must be internal', self)
                 return True
 
+        @cached_property
+        def it_catalog(self) -> Optional[CatalogName]:
+            if self.is_integration_test_catalog:
+                return None
+            else:
+                catalog = self.name + self._it_catalog_suffix
+                assert self._it_catalog_re.match(catalog), catalog
+                return catalog
```

Comment:
```suggestion
        @cached_property
        def it_catalog(self) -> 'Catalog':
            if self.is_integration_test_catalog:
                return self
            else:
                name = self.name + self._it_catalog_suffix
                assert self._it_catalog_re.match(name), name
                return self.catalogs[name]
```

See if a fixpoint like this and returning a Catalog instance simplifies things.

Reply:
Resolved at PL. We can't easily return instances of `Catalog`.